querqy-core/src/main/java/querqy/rewrite/commonrules/model/FilterInstruction.java

```diff
@@ -21,27 +21,37 @@
 public class FilterInstruction implements Instruction {
 
    final QuerqyQuery<?> filterQuery;
+    private final InstructionDescription instructionDescription;
 
-   public FilterInstruction(final QuerqyQuery<?> filterQuery) {
-      if (filterQuery == null) {
-         throw new IllegalArgumentException("filterQuery must not be null");
-      }
-      this.filterQuery = filterQuery instanceof BooleanQuery
-              ? InstructionHelper.applyMinShouldMatchAndGeneratedToBooleanQuery((BooleanQuery) filterQuery)
-              : filterQuery;
-   }
-
-   /* (non-Javadoc)
-    * @see querqy.rewrite.commonrules.model.Instruction#apply(querqy.rewrite.commonrules.model.PositionSequence, querqy.rewrite.commonrules.model.TermMatches, int, int, querqy.model.ExpandedQuery, java.util.Map)
+    @Deprecated // Use only for testing
+    public FilterInstruction(final QuerqyQuery<?> filterQuery) {
```

Comment:
Let's make it package visible then or remove it and change the tests

Reply:
I cannot make them package private as they are used by many different test classes & code (e.g. deprecated LineParser)

Reply:
(affects more or less all Instruction classes)

Reply:
We anyway know that we will change the structure of Instruction sooner or later